agents/check_mk_agent.linux

```diff
@@ -732,26 +732,35 @@ if [ -e /etc/openvpn/openvpn-status.log ]; then
         sed -e 1,3d -e '$d'
 fi
 
-is_timesync_succesful=false
+# Check if we're on a systemd based host
 if inpath systemctl; then
-    timesync_status=$(systemctl status ntp | awk '{if(NR==3) print $2}')
-    if [ $timesync_status == "active" ]; then
+    # We treat timesyncd separately
+    if inpath timedatectl; then
+       # Starting by getting the systemd version to ensure that 'timesync-status' is a supported arg
+        # Calling systemctl appears to be more portable across more systemd versions
+        systemd_version=$(systemctl --version | awk '{if(NR==1) print $2}')
+
+        # Only act if it's greater than or equal to version 239
+        if [ "${systemd_version}" -ge 239 ]; then
+            echo "<<<timesyncd>>>"
+            timedatectl timesync-status
+            stat -c %Y /var/lib/systemd/timesync/clock | awk '{print "[[["$1"]]]"}'
+        fi
+    fi
+
+    # Identify whether ntp/ntpd/openntpd is active
+    # shellcheck disable=SC2016
+    if [ "$(systemctl | awk '/ntp.service|ntpd.service/{print $3; exit}')" = "active" ]; then
```

Comment:
In our current implementation, we only want to retrieve the information from either systemd or ntp (not both) with systemd being preferred over ntp. Changing the if-statements would eventually resolve this.

Reply:
Hi Anthony,
Please see the function `section_ntp()` from the merged agent script here: https://github.com/rawiriblundell/checkMK/blob/merged_nix_agent/agents/check_mk_agent.merged#L1861

I've updated it to hopefully address what you're wanting to achieve.  Let me know if it seems to do what you want, and I can 'backport' it to the linux agent script.

Cheers
Rawiri

Reply:
Hi Rawiri, 

Thank you for your effort. I had a look through your section_ntp() and agree with your approach. I would like to add some further comments to conform it to our current version: 

- Chrony must be completely separate at this point, so only the ntp and systemd calls are necessary in the section_ntp() --> I will check/discuss the chrony implementation separately
- This was already mentioned by you: In the current iteration we would like to maintain the solaris and aix agents separately. In the last statement (if inpath ntpq) only the linux section would be currently required. A merged approach could be discussed in a different post

Thank you for your time and effort again :+1: 

Best, 

Anthony

Reply:
Hi Anthony,
I've just pushed a change with some adjustments made.

I've swapped calls to `mkecho()` to `echo`, as `mkecho()` is a function specific to the merged agent script.

I've kept Chrony within the function as it's time related, I've just moved it out of the systemd section and removed its function escape return.  So for every invocation of the agent, this function will be called.  And for every invocation of the function: if `chronyc` exists and `chronyd` appears to be active, it will run the chrony check, and this is independent of systemd and/or `ntpq` considerations.

I've also removed the AIX and Solaris code.

Cheers

Rawiri